### ISSUE TITLE
qiskit/visualization/state_visualization.py needs testing

### ISSUE THREAD

**Issue by 1ucian0** [Qiskit/qiskit#2306]
The code in `qiskit/visualization/state_visualization.py` needs testing. Current coverage is:

file | statements | missing | coverage
-- | -- | -- | --
qiskit/visualization/state_visualization.py | 392 | 369 | 6%

(This is a follow up from https://github.com/Qiskit/qiskit-terra/issues/300#issuecomment-489324685)

**Comment by 1ucian0** [comment]
Still a valid issue: https://coveralls.io/builds/24960860/source?filename=qiskit/visualization/state_visualization.py

**Comment by 1ucian0** [comment]
`on hold` until #2968 is decided.

**Comment by gabriellm1** [comment]
Is this still workable?

**Comment by 1ucian0** [comment]
@gabriellm1 it is! Probably following the https://github.com/Qiskit/qiskit-terra/pull/4544 style.

**Comment by abzsd** [comment]
can I work on this? This is my first issue.

**Comment by javabster** [comment]
hi @gabriellm1 are you working on this? If not I will assign to @abzsd

**Comment by gabriellm1** [comment]
Go ahead guys :)

**Comment by javabster** [comment]
thanks @gabriellm1 😄 @abzsd I'm assigning to you, let us know if you have any questions

**Comment by mrvee-qC** [comment]
# Update for Santa:elf:- QGoGP
✅ state_visualization.py is now 56.84% covered! (improvement over the 6% at start but a dedicated test would be great!) https://coveralls.io/builds/44969725/source?filename=qiskit%2Fvisualization%2Fstate_visualization.py

Will update comment/add more comments if cause is found/fixed!

> Python version 3.9.7
> qiskit-terra version: 0.19.1

**Comment by 1ucian0** [comment]
Did you have any change to have a look @abzsd ?

**Comment by smturro2** [comment]
Hi If @abzsd is no longer working on this issue, I'd love to be assigned and write the tests.

**Comment by javabster** [comment]
reassigned to @smturro2 due to no response from @abzsd 😄 Let us know if you need any assistance @smturro2! You can review the snapshot testing instructions in the [contributing guidelines](https://github.com/Qiskit/qiskit-terra/blob/main/CONTRIBUTING.md#snapshot-testing-for-visualizations) for guidance on mpl testing

**Comment by smturro2** [comment]
Hi @javabster thanks for assigning the issue. There seems to be tests in the snapshot testing for many of the functions in state_visualization. Was there specific tests in mind?

**Comment by javabster** [comment]
ah yes I see what you mean @smturro2! By checking the link to the coverage report I can see there are still quite a few lines (highlighted in red) that don't have test coverage. Some of these could be covered by more snapshot tests (e.g. adding another snapshot test with different arguments) and others could be covered by regular unit tests (e.g. I see a lot of the lines that raise errors are uncovered, it would be good to write unit tests that check the error is raised correctly). You don't need to reach 100% coverage, but I think if we could reach 80% coverage that would be great!

**Comment by jakelishman** [comment]
I don't think we actually catch the snapshot tests in the coverage report - all the snapshots are in `test/ipynb`, but the coverage report just runs `stestr run`, which only looks in `test/python`.  So we actually probably have slightly better coverage than it looks like we do, but we're not detecting it.

That said, we're definitely under-testing the visualisation modules, so any further efforts to help out on them are always welcome!

**Comment by javabster** [comment]
Hi @smturro2 are you still working on this issue?

**Comment by smturro2** [comment]
@javabster im no longer working on this👍🏻

**Comment by javabster** [comment]
ok thanks for letting me know, I'll unassign you so that others can jump in if they would like to work on it :)

**Comment by javabster** [comment]
sure thing @PSD-20, I'll assign to you! If you haven't contributed to terra before please read [the contributing guide](https://github.com/Qiskit/qiskit-terra/blob/main/CONTRIBUTING.md), and if you are new to open-source/github in general you might find [these videos ](https://www.youtube.com/watch?v=jj63uai3Qso&list=PLOFEBzvs-VvqKza-VLSYzWxnUMp-nO_aS&index=4) helpful.

**Comment by PSD-20** [comment]
Ohk. Also, Can you please assist me with how I should begin with this particular issue.

**Comment by javabster** [comment]
Hi @PSD-20 to get started with this you can see the coverage report for this file here: https://coveralls.io/builds/61022961/source?filename=qiskit%2Fvisualization%2Fstate_visualization.py

As you can see there are lots of lines of code highlighted in red, which means that they are lacking tests to cover them. So you would need to write some tests that would cover these lines of code. You can see examples of other tests for the visualisation module in `test/python/visualization`.

**Comment by RcrdPhysics** [comment]
Hi, is this issue open and not assigned, could I work on this issue?

**Comment by 1ucian0** [comment]
Current coverage is 57%. Not great, but not as bad as it used to be. So closing this issue. PR that increase test coverage are always welcomed.

<img width="445" height="128" alt="builds/79246978 qiskit/visualization/state_visualization.py" src="https://github.com/user-attachments/assets/6c23f383-9719-4861-99b6-8c6906257c2b" />
